dpe/src/commands/destroy_context.rs

```diff
@@ -26,7 +26,28 @@ impl CommandExecution for DestroyCtxCmd {
             return Err(DpeErrorCode::InvalidLocality);
         }
 
-        let to_destroy = (1 << idx) | dpe.get_descendants(context)?;
+        // mark consecutive retired parent contexts to be destroyed
+        let mut retired_contexts = 0u32;
+        let mut parent_idx = context.parent_idx as usize;
+        loop {
+            if parent_idx == Context::ROOT_INDEX as usize {
+                break;
+            } else if parent_idx >= dpe.contexts.len() {
+                return Err(DpeErrorCode::InternalError);
+            }
+            let parent_context = dpe.contexts[parent_idx];
+            if parent_context.state == ContextState::Retired {
```

Comment:
Take the following sequence of DPE commands:

```
parent_handle = 0xAB;

child_a = DeriveChild(parent, retain_parent = true);
child_b = DeriveChild(parent, retain_parent = true);
child_c = DeriveChild(parent, retain_parent = false);

DestroyContext(child_a);
```

The 3rd DeriveChild call would retire `parent`. This DestroyContext call would destroy `parent`, but it is still a parent of both `child_a` and `child_b`.

So I think we can only destroy it if it has no other children that aren't being destroyed.

Reply:
Good point, done.